core/lib/zksync_core/src/consensus/wrap_error.rs

```diff
@@ -0,0 +1,22 @@
+use std::fmt::Display;
+
+pub trait WrapError: Sized {
+    fn wrap<C: Display + Send + Sync + 'static>(self, c: C) -> Self {
+        self.with_wrap(|| c)
+    }
+    fn with_wrap<C: Display + Send + Sync + 'static, F: FnOnce() -> C>(self, f: F) -> Self;
+}
+
+impl WrapError for anyhow::Error {
```

Comment:
Nit: This implementation is used in 2 places (`impl WrapError for StorageError` and a single place in `put_block`), and can be replaced with `.context(_)`. IMO, it could make sense to specialize this extension for `StorageError` / `StorageResult` specifically and maybe rename methods to something like `db_context()` / `with_db_context()`.

Reply:
I'd rather prefer to generalize it, so that WrapError can be implemented for an arbitrary error type which accumulates context in a nontrivial way. Also StorageError ~ Canceled + Internal, a type which IMO is as applicable as anyhow::Error, so I think it would be useful to move StorageError to concurrency crate and rename it.

Reply:
anyhow::Error needs some helper to support WrapError, because (as mentioned above) it doesn't implement anyhow::Context directly. This helper doesn't necessarily have to be WrapError implementation, but it would look quite consistent IMO.

Reply:
my proposal: https://github.com/matter-labs/era-consensus/pull/43